api/server/models/room.js

```diff
@@ -175,6 +174,72 @@ module.exports = function (Room) {
             }
     );
     ///////////////////////////////////////////////////////////////////////////////////
+    ///////////////////////////////////////////////////////////////////////////////////
+    Room.list_my_rooms = function ( accessToken, currentTimestamp, callback) {
+        var User = Room.app.models.User;
+        User.relations.accessTokens.modelTo.findById(accessToken, function(err, accessToken) {
+            if( err ){
+                callback(null, 0, 'UnAuthorized', {});
+            }else{
+                if( !accessToken ){
+                    callback(null, 0, 'UnAuthorized', {});
+                }else{
+                    var userId = accessToken.userId
+                    userId = new ObjectID( userId );
+                    Room.find({
+                        "where": {room_users : {'in':[userId]}},
+                        "include": [{
+                            relation: 'room_owner', 
+                            scope: {
+                                fields: ['name'],
+                            }
+                        },{
+                            relation: 'room_users', 
+                            scope: {
+                                fields: ['name'],
+                            }
+                        }]
+                    },function (err, result) {
+                        if( err ){
+                            console.log( err );
+                            callback(null, 0, 'try again', {});
+                        }else{
+                            if( result.length > 0 ){
+                                var data = {
+                                    'rooms' : result
+                                };
+                                callback( null, 1, 'Rooms found', data );
```

Comment:
data will have 'rooms'.
in future if some more info to be send like number of rooms etc, we can add this to data.